### PR TITLE
Fix/add dynamic config

### PR DESCRIPTION
**Pull request by avpash43**
add dynamic config

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 ---
+zk_config: []
 zk_version: 3.4.14
+zk_config_file_list: >-
+  {{ (zk_version is version_compare('3.5.0', '>=')) | ternary(
+    ['zoo.cfg', 'log4j.properties', zoo_cfg_dynamic],
```
Comment (ViachaslauKabak) [inline defaults/main.yml]: in template zoo.cfg.dynamic

File: templates/zoo.cfg.dynamic.j2 [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{%- set ips = [] %}
+{%- for host in groups[zk_inventory_group] %}
+{{- ips.append(dict(id=loop.index, host=host, ip=hostvars[host]['ansible_default_ipv4'].address)) }}
+{%- endfor %}
+
+{% for server in ips %}
+{% if server.host is defined %}
+{% if server.ip is defined %}
+server.{{server.id}}={{server.ip}}:2888:3888;{{ zk_client_port }}
```
Comment (ViachaslauKabak) [inline templates/zoo.cfg.dynamic.j2]: need to add server role

Comment (ViachaslauKabak) [reply templates/zoo.cfg.dynamic.j2]: ad add nothing if not set

File: defaults/main.yml [hunk 3]
```diff
@@ -13,10 +19,12 @@ zk_tick_time: 2000
 zk_autopurge_purgeInterval: 0
 zk_autopurge_snapRetainCount: 10
 # 3.5.0 and later
+zk_reconfig_enabled: 'true'
```
Comment (ViachaslauKabak) [inline defaults/main.yml]: Add to readme

File: defaults/main.yml [hunk 4]
```diff
@@ -13,10 +19,12 @@ zk_tick_time: 2000
...
 zk_standalone_enabled: 'true'
```
Comment (ViachaslauKabak) [inline defaults/main.yml]: check if in readme
add to  templates/zoo.cfg.j2 for version > 3.5.0

File: templates/zoo.cfg.j2 [hunk 5]
```diff
@@ -28,3 +28,13 @@ server.{{loop.index}}={{server}}:2888:3888
 {% endif %}
 {% endfor %}
 {% endif %}
+
+{% if zk_config | length %}
+{{ zk_config | to_nice_yaml | regex_replace(': ','=')}}
+{% endif %}
+
+{% if (zk_version is version_compare('3.5.0', '>=')) %}
+dynamicConfigFile={{ zk_dir }}/conf/zoo.cfg.dynamic
```
Comment (ViachaslauKabak) [inline templates/zoo.cfg.j2]: it will not fit the package install

Comment (ViachaslauKabak) [reply templates/zoo.cfg.j2]: zk_package_config_dir for package install

File: molecule/cloud-epc-delegated/playbook.yml [hunk 6]
```diff
@@ -5,3 +5,5 @@
     - role: lean_delivery.java
     - role: ansible-role-zookeeper
       zk_force_myid: False
+      zk_config:
```
[the remark this answers is not in the document]
Comment (ViachaslauKabak) [reply molecule/cloud-epc-delegated/playbook.yml]: any news?

File: templates/zoo.cfg.dynamic.j2 [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+{% for server in ips %}
+{% if server.host is defined %}
+{% if server.ip is defined %}
+server.{{server.id}}={{server.ip}}:2888:3888:participant;{{ zk_client_port }}
+{% else %}
+server.{{server.id}}={{server.host}}:2888:3888:participant;{{ zk_client_port }}
+{% endif %}
+{% else %}
+server.{{loop.index}}={{server}}:2888:3888:participant;{{ zk_client_port }}
```
Comment (ViachaslauKabak) [inline templates/zoo.cfg.dynamic.j2]: why we cannot choose what type server will be?
observer is the type too

File: molecule/cloud-epc-delegated/playbook.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (avpash43) [reply molecule/cloud-epc-delegated/playbook.yml]: Playbook of default test has the lines:
      zk_config:
          maxClientCnxns: 60

[the remark this answers is not in the document]
Comment (ViachaslauKabak) [reply molecule/cloud-epc-delegated/playbook.yml]: I see then default/main.yml has zk_config: []

**Pull request merged**